### PR TITLE
V4

### PR DESCRIPTION
**Pull request by thgreasi**
Change-type: major

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "README.md"
   ],
   "engines": {
-    "node": ">=16.13.0"
+    "node": ">=18.19.1"
```
Comment (otaviojacobi) [inline package.json]: @thgreasi any reason why you picked 18.19.1? cli is currently on ^18.6.0

Comment (thgreasi) [reply package.json]: Oh that's a good point. It seems that I just picked the latest I had locally given that I was doing a bump, but forgot to check the CLI. Nice catch :+1:

Comment (thgreasi) [reply package.json]: Oh wait @otaviojacobi , the CLI is on `^20.6.0`. That's what you meant right?

Comment (thgreasi) [reply package.json]: fwiw, afair I chose v18 b/c that was the minimum that supports native fetch w/o flags.

Comment (otaviojacobi) [reply package.json]: :man_facepalming: I did misread it and though we were already bumping this to 20. Please disregard my comment

**Comment by otaviojacobi** [comment]
We might also want a check from supervisor devs in this? https://github.com/balena-os/balena-supervisor/blob/master/package.json#L42

**Comment by pipex** [comment]
The supervisor only uses balena-compose for setting up its development image @otaviojacobi. This change seems to be related to the typing improvements on Pine, I don't think that will affect the supervisor but I'll check it out. Thanks for bringing it to my attention!

**Comment by otaviojacobi** [comment]
hey @pipex I think the typing pr is a follow up to this, but the more "breaking" thing here that could in impact is in fact this commit: https://github.com/balena-io-modules/balena-compose/pull/52/commits/40da06d268ac1c398bdc1747f9577a97c8841a83 it just effectively expects the engine to break with v1 manifests (no changes on balena-compose itself, but we should be aware that this will happen on more recent docker releases)

Also important is probably this: https://github.com/balena-io-modules/balena-compose/pull/52/commits/8ce4ee1da12ced9b6320d6650befcf64d8749392 we are using native fetch api to talk to the registry instead of [request library](https://github.com/request/request) which could potentially change something as fetch http client is far lighter/less feature full (but so far we haven't seen any change)

File: lib/release/api.ts [hunk 2]
```diff
@@ -1,42 +1,12 @@
...
-	/**
-	 * The complete string to forward as Authorization HTTP header, eg.
-	 * `Bearer <authtoken>`.
-	 */
-	auth: string;
-
-	retry?: RetryParameters;
-}
-
-export function createClient(config: ClientConfig): PinejsClientRequest {
```
Comment (dfunckt) [inline lib/release/api.ts:28]: Now that this is gone, there is a comment at line 58/28 that needs to be updated.

**Review by dfunckt — COMMENTED** [review]
It all looks good to me.

**Comment by thgreasi** [comment]
@otaviojacobi are we fine w/ this? Are we just pending an approval from someone?

**Comment by otaviojacobi** [comment]
@thgreasi I have tested pushing to a local device, pushing to cloud, builiding & deploying to cloud too. 
More over, I have isolated and test the bit where you replaced `request` for `fetch` and it seems this bit is just some very specific thing for resolving node docker image versions from old `resin` balena images. I manually tested and all was the exact same result as with request lib so I don't think there are any blockers (besides that, balena-cli tests also passed locally)

So IMHO, yes, we are just missing a review/approval, please notice that I also ammended one of your commits to address Akis's comment.

**Comment by pipex** [comment]
Thanks for the info @otaviojacobi. This should not cause any issues on the supervisor the supervisor as this module is not used in production, it is only used to build supervisor images for testing the supervisor on a live device ([npm run sync](https://github.com/balena-os/balena-supervisor/blob/master/package.json#L26))

**Comment by thgreasi** [comment]
Cool @otaviojacobi I had tested exactly those things as well :+1:  (The node resolver change ended up being the hardest thing to test)

**Review by otaviojacobi — APPROVED** [review]

**Pull request merged**